solidity/random-beacon/README.adoc

```diff
@@ -177,103 +166,122 @@ When the hard timeout is reached, anyone can notify about this fact by calling
 we consider this approach safe and good enough. An important advantage of this
 approach is that honest players can decide off-chain when it makes sense to
 submit an operator inactivity claim and mark someone as ineligible for rewards.
 For example, marking an operator ineligible for rewards for the next two weeks
 have a higher impact than prolonging reward ineligibility for 10 minutes for an
 operator that was already marked as ineligible for rewards. This approach does
 not increase the gas cost of a happy path and leaves some freedom to group
-members. They may mark as ineligible operators who turned off their nodes,
+members. They can mark as ineligible operators who turned off their nodes,
 operators whose nodes never participate in signing because they are
 misconfigured, or operators who notoriously miss their turn in submitting relay
-entry.
+entries.
+
+=== Rewards
+
+T rewards are allocated to all operators registered in the beacon sortition
+pool, excluding operators who were marked as ineligible for rewards as a result
+of <<inactivity, inactivity>>. Rewards are allocated proportionally to the
+operator's weight in the pool. 
+
+[[transaction-incentives]]
+=== Transaction Incentives
+
+There are two types of transactions: <<operator-only,Operator-Only>>,
+<<public-knowledge,Public-Knowledge>>, and <<punishment,Punishment>>.
+
+[[operator-only]]
+==== Operator-Only
+Operator-Only transactions are where only the operators have access to the
+information required to assemble the transaction with the right input
+parameters.
+
+In order to avoid all operators racing to submit the transaction at the same
+time, we have an off-chain informal agreement to submit based on the operator's
+position in the group (can use the hash of the group's pubkey).
+
+If the designated operator does not submit their transaction before a timeout
+expires, the duty moves to the next operator and the group can sign a
+transaction to mark that operator as inactive. Since there is no slashing
+reward, and since this transaction can only be submitted by an operator, this
+transaction is also Operator-Only.
+
+In order to compensate the operator for posting the transaction, the gas spent
+will be reimbursed by a DAO-funded eth pool in the same transaction (see
+`Reimbursable` and `ReimbursementPool` smart contracts).
+
+Operator-only transactions are `submitDkgResult`, `submitRelayEntry`,
+`notifyOperatorInactivity`, and `approveDkgResult` for a certain number of
+blocks, before a timeout for the original DKG result submitter to call this
+function elapses.
+
+[[public-knowledge]]
+==== Public-Knowledge
+Public-Knowledge transactions are where anyone has access to the information
+required to assemble the transaction and the transaction does not lead to
+punishment.
+
+In order to prevent wasting gas on racing to submit, such transactions need to
+be executed rarely, and off-chain clients should follow the informal agreement
+about the submission order.
```

Comment:
We don't have the idea of 'trusted maintainers' in Random Beacon, right? Do we already have an idea how the clients will decide whose order it is to post the tx?

Reply:
We do not need them for Random Beacon. The only "maintainer" transaction is `notifyDkgTimeOut` and it should not be called often (hopefully never called), so we do not have to worry about the order of transaction submitters. In tBTC, we have submit sweep proof, submit redemption proof that are called *very often* and there, we need to worry about the order maintainers should follow.